server/artifacts/artifact_server.go

```diff
@@ -128,6 +128,9 @@ func (a *ArtifactServer) returnArtifact(ctx context.Context, w http.ResponseWrit
 	kubeClient := auth.GetKubeClient(ctx)
 
 	art := wf.Status.Nodes[nodeId].Outputs.GetArtifactByName(artifactName)
+	if art == nil {
```

Comment:
What if inputs and outputs both contain artifact with the same name? Then we would only get the output artifact.

Reply:
Yes. That's why I propose a different approach in https://github.com/argoproj/argo-workflows/pull/5705#discussion_r615542424, to differentiate input/output from the URL.

Will implement it once we reach an agreement. Thanks!